FirebaseSessions/ProtoSupport/Protos/sessions.proto

```diff
@@ -0,0 +1,160 @@
+  // The collection status of the FirePerf SDK.
+  DataCollectionState performance = 1;
+  // The collection status for the Crashlytics SDK.
+  DataCollectionState crashlytics = 2;
+}
+
+// Enum denoting all possible states for SDK data collection.
+enum DataCollectionState {
+  COLLECTION_UNKNOWN = 0;
+  // This product SDK is not present in this version of the app.
+  COLLECTION_SDK_NOT_INSTALLED = 1;
+  // The product SDK is present and collecting all product-level events.
+  COLLECTION_ENABLED = 2;
+  // The product SDK is present but data collection for it has been locally
+  // disabled.
+  COLLECTION_DISABLED = 3;
+  // The product SDK is present but data collection has been remotely disabled.
+  COLLECTION_DISABLED_REMOTE = 4;
+  // Indicates that the product SDK is present, but session data is being
+  // collected, but the product-level events are not being uploaded.
+  COLLECTION_SAMPLED = 5;
+}
+
+// App-level information collected from the device.
+message ApplicationInfo {
+  // Commonly known as the GMP App Id
+  string app_id = 1;
+  // The manufacturer supplied device model name. Eg. 'N5502L'
+  string device_model = 2;
+  // The development platform used by the app. Eg. 'Unity', 'Flutter'
+  string development_platform_name = 3;
+  // The version of the development platform in use by the app. Eg. 2020.3.33f1
+  string development_platform_version = 4;
+
+  oneof platform_info {
+    // App info relevant only to Android apps
+    AndroidApplicationInfo android_app_info = 5;
+    // App info relevant only to Apple apps
+    AppleApplicationInfo apple_app_info = 6;
+  }
+}
+
+// Android-specific app information
+// In addition to these fields, the Firelog SDK also collects:
+// - application_build
+// - mcc_mnc
+// - sdk_version(app's build version)
+// - manufacturer
+// - model
+// - country
+// - network_connection_info
+message AndroidApplicationInfo {
+  // The applications package name. Eg. com.google.search
+  string package_name = 1;
+  // The version of the Android Firebase-Sessions SDK in use
+  string sdk_version = 2;
+  // The version name as specified in the app's manifest
+  // https://developer.android.com/guide/topics/manifest/manifest-element#vname
+  string version_name = 3;
+}
+
+// Apple-specific app information
+// In addition to these fields, the Firelog SDK also collects:
+// - application_build
+// - application_version
+// - os_full_version
+// - model
+// - country
+message AppleApplicationInfo {
+  // The application's bundle id. Eg. com.google.search
+  string bundle_short_version = 1;
+  // The version of the Apple Firebase-Sessions SDK in use
+  string sdk_version = 2;
+  // Describes the network connectivity of the device
+  NetworkConnectionInfo network_connection_info = 3;
+  // Collects the OS running. Eg. iOS, iPadOs, tvOS, etc.
+  OsName os_name = 4;
+}
+
+// The name of the operating system on which the application is running.
+// Currently Apple only.
+enum OsName {
```

Comment:
We don't currently have plans to split out Catalyst and Mac. These are shown as filters in the Crashlytics dashboard, and we felt that you as a developer would know whether your app was a macOS app or an iOS app running on mac (because by bundle ID, your app can only be one or the other IIRC).

![Screen Shot 2022-10-13 at 4 07 29 PM](https://user-images.githubusercontent.com/555046/195698838-ada7abae-1
![Screen Shot 2022-10-13 at 4 07 26 PM](https://user-images.githubusercontent.com/555046/195698886-2edfd6db-51b3-4711-aba5-d09ee73605e2.png)
54e-4c9b-87c4-9afc9c1d9cd1.png)